chapter_4/ex04_directory/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# CMakeLists file for the Chapter 4 Example nr. 04: 
+# Install directories
+#
+# SPDX-License-Identifier: MIT
+
+project(
+  ch4_ex04_directory
+  VERSION 1.0
+  DESCRIPTION "Chapter 4 Example 04, install directory"
+  LANGUAGES CXX)
+
+install(DIRECTORY dir1 TYPE LOCALSTATE FILES_MATCHING PATTERN "*.x") 
+install(DIRECTORY dir2 TYPE LOCALSTATE FILES_MATCHING PATTERN "*.hpp" EXCLUDE PATTERN "*") 
+install(DIRECTORY dir3 TYPE LOCALSTATE PATTERN "bin" EXCLUDE) 
```

Comment:
Prefer to use the `DESTINATION` form of the command.

Reply:
Is there a specific reason behind this suggestion?

Reply:
It's way less common than the variable form and it's better to have `GNUInstallDirs` included, so the variables could be defined on the command line as well. However, now that I think about it, if you combine this with setting the `LANGUAGES` to `NONE` then you must also set the `CMAKE_INSTALL_LIBDIR` variable before including `GNUInstallDirs`, because it can't detect the platform without a language enabled. See [example](https://github.com/friendlyanon/cmake-init-header-only/blob/master/cmake/install-rules.cmake#L6).

In that case, do as you see fit.

Reply:
IMO the case where `LANGUAGES` is set to `NONE` is an edge case. I usually also use the `DESTINATION` with `GNUInstallDirs` but for no special reason, except that I learned it that way...

Reply:
`LANGUAGES NONE` is not an edge case. This project doesn't build anything, much like a header-only library. However, `GNUInstallDirs` requires either a language enabled or `CMAKE_INSTALL_LIBDIR` defined to not fail the configuration.